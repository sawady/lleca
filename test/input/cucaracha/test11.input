
fun main() {
  h := 104
  o := 111
  l := 108
  a := 97
  enter := 10
  putChar(h)
  putChar(o)
  putChar(l)
  putChar(a)
  putChar(enter)
}
