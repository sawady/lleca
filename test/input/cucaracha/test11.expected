Cons(
  Function(
    main,
    Nil,
    Unit,
    Cons(
      StmtAssign(
        h,
        ExprConstNum(
          104
        )
      ),
      Cons(
        StmtAssign(
          o,
          ExprConstNum(
            111
          )
        ),
        Cons(
          StmtAssign(
            l,
            ExprConstNum(
              108
            )
          ),
          Cons(
            StmtAssign(
              a,
              ExprConstNum(
                97
              )
            ),
            Cons(
              StmtAssign(
                enter,
                ExprConstNum(
                  10
                )
              ),
              Cons(
                StmtCall(
                  putChar,
                  Cons(
                    ExprVar(
                      h
                    ),
                    Nil
                  )
                ),
                Cons(
                  StmtCall(
                    putChar,
                    Cons(
                      ExprVar(
                        o
                      ),
                      Nil
                    )
                  ),
                  Cons(
                    StmtCall(
                      putChar,
                      Cons(
                        ExprVar(
                          l
                        ),
                        Nil
                      )
                    ),
                    Cons(
                      StmtCall(
                        putChar,
                        Cons(
                          ExprVar(
                            a
                          ),
                          Nil
                        )
                      ),
                      Cons(
                        StmtCall(
                          putChar,
                          Cons(
                            ExprVar(
                              enter
                            ),
                            Nil
                          )
                        ),
                        Nil
                      )
                    )
                  )
                )
              )
            )
          )
        )
      )
    )
  ),
  Nil
)
